src/main/java/net/javadiscord/javabot/data/h2db/message_cache/dao/MessageCacheRepository.java

```diff
@@ -0,0 +1,164 @@
+package net.javadiscord.javabot.data.h2db.message_cache.dao;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.javadiscord.javabot.data.h2db.message_cache.model.CachedMessage;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Dao class that represents the QOTW_POINTS SQL Table.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class MessageCacheRepository {
+	private final Connection con;
+
+	/**
+	 * Inserts a new {@link CachedMessage} object.
+	 *
+	 * @param message The Message to insert.
+	 * @return Whether there were rows affected by this process.
+	 * @throws SQLException If an error occurs.
+	 */
+	public boolean insert(CachedMessage message) throws SQLException {
+		PreparedStatement stmt = con.prepareStatement("INSERT INTO message_cache (message_id, author_id, message_content) VALUES (?, ?, ?)",
+				Statement.RETURN_GENERATED_KEYS
+		);
+		stmt.setLong(1, message.getMessageId());
+		stmt.setLong(2, message.getAuthorId());
+		stmt.setString(3, message.getMessageContent());
+		int rows = stmt.executeUpdate();
+		stmt.close();
+		return rows > 0;
+	}
```

Comment:
Apply this to all other methods in this class
```suggestion
		try (PreparedStatement stmt = con.prepareStatement("INSERT INTO message_cache (message_id, author_id, message_content) VALUES (?, ?, ?)",
				Statement.RETURN_GENERATED_KEYS
		)) {
	    	stmt.setLong(1, message.getMessageId());
	    	stmt.setLong(2, message.getAuthorId());
	    	stmt.setString(3, message.getMessageContent());
		    int rows = stmt.executeUpdate();
		    stmt.close();
		    return rows > 0;
   }
	}
```